ci/build-msys2.sh

```diff
@@ -1,7 +1,9 @@
 #!/bin/sh -e
 
 if [ "$1" = "meson" ]; then
-    meson setup build            \
+    python3.11 -m ensurepip
+    python3.11 -m pip install meson
+    python3.11 -m mesonbuild.mesonmain setup build \
```

Comment:
This looks like it could be easier/more portably done by using a new venv with `python3.11`

Reply:
venv creation is slow(er), we don't want to do that. (also it doesn't work currently)

Reply:
Ok, there was any update on MSYS2. venv now works I can use that.

Reply:
Venv creation is probably slower because it installs a copy of pip and setuptools. You can pass an option to disable this (and another option to enable use of the system site-packages so that the venv does have access to `venv/bin/python -m pip`). This should make venv creation almost instantaneous.

Reply:
It is fast enough in fact. My main issue were that it was completely not working with 3.11.2-1, but they updated to 3.11.2-2 and seems to work again.